### PR TITLE
API for custom descriptors and custom isApplicable

### PR DESCRIPTION
**Pull request by rhagelstrom**
Curious on thoughts on this.
I have a couple of more ideas that aren't related to this but rebased on top of this

### CODE REVIEW

**Comment by bmos** [comment]
I've only skimmed it but I like the approach.
This could lead to some really interesting places!

**Comment by rhagelstrom** [comment]
The big take away is to keep all the descriptors organized so we don't get to the point of searching through a bazillion different cases on things that will never match.

File: scripts/manager_aura_api.lua [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+    for _, rTable in pairs(aCustomDescriptors) do
+        if rTable.fCustomApplicable and not rTable.fCustomApplicable(nodeEffect, rSource, rTarget, rDetails) then
+            bReturn = false
+            break
+        end
+    end
+    return bReturn
+end
+
+-- match on 6 char hex string
```
Comment (mattekure) [inline scripts/manager_aura_api.lua]: isHex() needs to match on an 8 character hex string.  the FG color strings include alpha at the beginning, so its AARRGGBB.

**Comment by mattekure** [comment]
I also like it.  I can update to use this registration easily.  I did make a note that the hex matching needs to match 8 chars not 6 due to the alpha at the beginning.  AARRGGBB.

File: scripts/manager_aura_api.lua (hunk 1 above)
Comment (rhagelstrom) [reply scripts/manager_aura_api.lua]: Nice catch. updated

**Comment by rhagelstrom** [comment]
> I also like it. I can update to use this registration easily. I did make a note that the hex matching needs to match 8 chars not 6 due to the alpha at the beginning. AARRGGBB.

I think for the custom aura images, you'll want to preface all those with an identifier, so like  'av fog cloud' so they can partial match and be kept out of falling through to the aOther category.

**Pull request merged**